test/functional/mempool_persist.py

```diff
@@ -30,26 +30,20 @@
     does not overwrite a previously valid mempool stored on disk.
 
 """
+import time
 
+from test_framework.mininode import wait_until
 from test_framework.test_framework import BitcoinTestFramework
 from test_framework.util import *
 
 class MempoolPersistTest(BitcoinTestFramework):
 
     def __init__(self):
         super().__init__()
+        # We need 3 nodes for this test. Node1 does not have a persistent mempool.
         self.num_nodes = 3
         self.setup_clean_chain = False
-
-    def setup_network(self):
-        # We need 3 nodes for this test. Node1 does not have a persistent mempool.
-        self.nodes = []
-        self.nodes.append(start_node(0, self.options.tmpdir))
-        self.nodes.append(start_node(1, self.options.tmpdir, ["-persistmempool=false"]))
-        self.nodes.append(start_node(2, self.options.tmpdir))
-        connect_nodes_bi(self.nodes, 0, 2)
-        connect_nodes_bi(self.nodes, 1, 2)
-        self.is_network_split = False
+        self.extra_args = [[], ["-persistmempool=false"], []]
```

Comment:
Our parser does not support string representations of booleans. You should only use integers here...